File: 1201-1300/1207-unique-occurence.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        # count the occurence of each number
        num_dict = {}
        for num in arr:
            if num in num_dict:
                num_dict[num] += 1
            else:
                num_dict[num] = 1
                
        # Remember that the keys in num_dict are unique
        # if the count of values of dict is the same as the count
        # of the keys(unique), it also means the number of occurence is unique as well
        return len(num_dict) == len(set(num_dict.values()))

File: 1201-1300/1207-unique-occurence.java
// O(n), 1ms, 37.2MB
class Solution {
  public boolean uniqueOccurrences(int[] arr) {
      HashMap<Integer, Integer> map = new HashMap();
      
      for(int num : arr) {
          map.put(num, map.getOrDefault(num, 0) + 1);
      }
      
      HashSet<Integer> set = new HashSet();
      for(Integer count : map.values()) {
          if(!set.add(count)) {
              return false;
          }
      }
      
      return true;
  }
}